allenact/algorithms/onpolicy_sync/vector_sampled_tasks.py

```diff
@@ -926,6 +926,12 @@ def _task_sampling_loop_generator_fn(
                                 step_result = step_result.clone({"info": {}})
                             step_result.info[COMPLETE_TASK_METRICS_KEY] = metrics
 
+                        task_callback_data = current_task.task_callback_data()
```

Comment:
I would prefer if tasks didn't know about callbacks as this creates a bidirectional dependency between the tasks and callbacks. Could we instead pass a function (or callable object) to the `VectorSampledTask` and then do something like `task_callback_data = task_callback_data_fn(current_task)`. This `task_callback_data_fn` could be returned by the experiment config similarly as to how `make_sampler_fn` is.

Reply:
Actually even better would be if each `Callback` was required to have this function defined on itself as a static method and we just passed in a list of these methods (that way the experiment config wouldn't need to know which callbacks were going to be used with it making it easier to mix and match from the command line).

Reply:
That’s an interesting idea! Like an `on_task_end(task: Task)` callback.